app/controllers/menus_controller.rb

```diff
@@ -2,6 +2,11 @@ class MenusController < ApplicationController
 
   def show
     @restaurant = Restaurant.find(params[:restaurant_id])
-    @cart_nr = ShoppingCartItem.count
+    if session[:cart_id]
+      @cart = ShoppingCart.find(session[:cart_id])
+      @cart_nr = @cart.shopping_cart_items.count
```

Comment:
okay.. but I still don't think defining `@cart_nr` is necessary to achieve that. Not a blocker for me though if it's out of scope for this feature.